### DISCUSSION TITLE
Programmazione accessione a giorni alterni

### DISCUSSION THREAD

**Discussion by array81** [indomus/forum#2549] [HUB: Home Assistant]
Ho creato una dispositivo per il controllo dell'irrigazione del mio giardino. Con ESPHome vado di fatto a aprire le valvole di zona e a accendere la pompa.
Ora vorrei programmare l'apertura di una valvola in particolare vorrei che venisse aperta la mattina alle 7.30 per 10 minuti un giorno si e uno no. Il problema sono i giorni alterni. Ho valutato sia di usare il calendario integrato in HA, sia l'aiutante Calendarizzazione, sia Scheduler Component. In tutti i casi tuttavia la programmazione non permette di settare un giorno si e uno no. Qualche suggerimento a riguardo?

**Comment by 50494554524F** [comment]
due idee che mi vengono al volo... puoi usare un contatore o differenza tra now() e data di riferimento modulo 2, messi in un binary_sensor da usare come trigger

**Comment by KC-inDomus** [comment]
> Ho creato una dispositivo per il controllo dell'irrigazione del mio giardino. Con ESPHome vado di fatto a aprire le valvole di zona e a accendere la pompa. Ora vorrei programmare l'apertura di una valvola in particolare vorrei che venisse aperta la mattina alle 7.30 per 10 minuti un giorno si e uno no. Il problema sono i giorni alterni. Ho valutato sia di usare il calendario integrato in HA, sia l'aiutante Calendarizzazione, sia Scheduler Component. In tutti i casi tuttavia la programmazione non permette di settare un giorno si e uno no. Qualche suggerimento a riguardo?

io la farei facile: creerei un input_boolean che alla mezzanotte, tramite un'automazione ad hoc banalissima, inverta il suo stato on/off. poi creerei l'automazione principale che venga eseguita solo quando lo stato di input_boolean sia ON (off OFF, a scelta)

**Comment by array81** [comment]
Si, effettivamente così funziona. Sinceramente credevo che una casistica come la mia fosse già trattata direttamente da HA invece mi pare di capire che la schedulazione degli eventi sia ancora un po' rigida. Alla fine eventi del tipo, un giorno si e uno no oppure uno si e due no sono abbastanza comuni.
Comunque grazie degli spunti.

**Comment by zanac** [comment]
Io ho fatto una template condition, chiaramente con questa i mesi con 31 va due volte di seguito (nel mio caso non è un problema

{{ now().strftime("%-d") | int % 2 == 0  }}
